## Задание 1

### Критерий $\chi^2 $

In [1]:
import pandas as pd
import numpy as np
from scipy import stats
data1 = pd.read_excel('data.xlsx')
Xy = [i.values for i in data1['Weight'],data1['Distanse']]
def modchisq(Xy):
    chi2, prob, df, expected = stats.chi2_contingency(Xy)
    output = "test Statistics: {}\ndegrees of freedom: {}\np-value: {}\n"
    print (output.format(chi2, df, prob))
    print (expected)

In [2]:
data1

Weight  Distanse
0     67.3      8828
1     73.1      8332
2     72.6      7101
3     73.9      8127
4     74.7      5985
5     75.0     10827
6     77.1      8565
7     90.4      4295
8     70.9      4750
9     74.7      6379
10    80.2     10103

In [3]:
def chisq(Xy):
    tmp=[Xy[0][i]+Xy[1][i] for i in range(len(Xy[0]))]
    s=sum(tmp)
    tmp=[i/float(s) for i in tmp]
    expected=[[i*sum(Xy[j]) for i in tmp]for j in range(2)]
    rez=[[((Xy[j][i]-expected[j][i])**2)/expected[j][i] for i in range(len(Xy[0]))]for j in range(2)]
    print (expected)
    return sum(rez[0])+sum(rez[1])
chisq(Xy)

[[87.756095261757054, 82.920053993074347, 70.770758149780264, 80.905530070053103, 59.781638669597328, 107.55308427413077, 85.258164520772837, 43.263923663160242, 47.560325075872036, 63.668624103830275, 100.46180221797179], [8807.5439047382442, 8322.1799460069269, 7102.8292418502197, 8119.9944699299476, 5999.9183613304021, 10794.446915725872, 8556.8418354792284, 4342.1360763368393, 4773.3396749241283, 6390.0313758961693, 10082.73819778203]]


90.642063578454525

In [4]:
modchisq(Xy)

test Statistics: 90.6420635785
degrees of freedom: 10
p-value: 3.99462573305e-15

[[    87.75609526     82.92005399     70.77075815     80.90553007
      59.78163867    107.55308427     85.25816452     43.26392366
      47.56032508     63.6686241     100.46180222]
 [  8807.54390474   8322.17994601   7102.82924185   8119.99446993
    5999.91836133  10794.44691573   8556.84183548   4342.13607634
    4773.33967492   6390.0313759   10082.73819778]]


При уровне значимости $ \alpha = (1-0,05)$, и $k=(11-1)(2-1)=10$ => $\chi^2_{spec}=90.642063578454525 $, a $\chi^2_{k=10,\alpha=0.95}=3.994$ => Так как $\chi^2_{spec} > \chi^2_{k=10,\alpha=0.95}$ - есть основание отвергнуть гипотезу о зависимости данных велечин. 

### Т-критерий Стьюдента

In [5]:
def modtks(Xy):
    t, prob = stats.ttest_ind(Xy[0], Xy[1])
    output = "test Statistics: {}\np-value: {}\n"
    print (output.format(t, prob))

In [6]:
def tks(Xy):
    avxy = [np.average(Xy[0]), np.average(Xy[1])]
    serror=[[((Xy[j][i]-avxy[j])**2) for i in range(len(Xy[0]))]for j in range(2)]
    Sd=np.sqrt((sum(serror[0])+sum(serror[1]))/((len(Xy[0])-1)*(len(Xy[0]))))
    return np.abs(avxy[0]-avxy[1])/Sd
tks(Xy)

11.946226943133036

In [7]:
modtks(Xy)

test Statistics: -11.9462269431
p-value: 1.47178838936e-10



При уровне значимости $\alpha=0.05$ и $k=(11+11)-2=20$ => $t_{spec}=11.946226943133036 $, a $t_{k=20,\alpha=0.05}=2.09$

2.09 для P $ \le $ 0.05

2.85 для P $ \le $ 0.01

3.85 для P $ \le $ 0.001

Так как $t_{spec} > t_{k=10,\alpha=0.05}$ - есть основание отвергнуть гипотезу о зависимости данных велечин. 

## Задание 2

In [8]:
data2 = pd.read_excel('data2.xlsx')
Xy2 = [i.values for i in data2['Test1'],data2['Sumup']]

In [9]:
chisq(Xy2)

[[7.5671794871794864, 26.039999999999999, 12.463589743589743, 27.375384615384615, 9.5702564102564089, 25.14974358974359, 23.369230769230771, 25.594871794871793, 24.704615384615384, 12.241025641025642, 22.924102564102562], [26.432820512820513, 90.959999999999994, 43.536410256410257, 95.624615384615382, 33.429743589743588, 87.850256410256407, 81.630769230769232, 89.405128205128207, 86.29538461538462, 42.758974358974356, 80.075897435897431]]


17.956288757244966

In [10]:
tks(Xy2)

6.3315791369721133

## Задание 3

In [11]:
data3 = pd.read_excel('data3.xlsx')
Xy3 = [i.values for i in data3['CREA'],data3['TG']]
stats.spearmanr(Xy3[0],Xy3[1])

SpearmanrResult(correlation=0.30029461361660664, pvalue=0.01974141390904845)